From f5fb997e0b19863d9b865882fa81f90ae67c7e03 Mon Sep 17 00:00:00 2001
From: Sai Vemuri <svemuri@redhat.com>
Date: Thu, 11 Sep 2014 22:42:42 -0400
Subject: [infiniband] cxgb4: Fix underflows in c4iw_create_qp()

Message-id: <1410475447-94392-24-git-send-email-svemuri@redhat.com>
Patchwork-id: 93873
O-Subject: [RHEL7.1 PATCH BZ 1124947 023/108] RDMA/cxgb4: Fix underflows in c4iw_create_qp()
Bugzilla: 1124947
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

These sizes should be unsigned so we don't allow negative values and
have underflow bugs.  These can come from the user so there may be
security implications, but I have not tested this.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit ff1706f4feb8e0e1a2e56a8dd57e17a4b45649b5)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/cxgb4/qp.c b/drivers/infiniband/hw/cxgb4/qp.c
index 3b62eb5..ff1b671 100644
--- a/drivers/infiniband/hw/cxgb4/qp.c
+++ b/drivers/infiniband/hw/cxgb4/qp.c
@@ -1529,7 +1529,7 @@ struct ib_qp *c4iw_create_qp(struct ib_pd *pd, struct ib_qp_init_attr *attrs,
  struct c4iw_cq *schp;
  struct c4iw_cq *rchp;
  struct c4iw_create_qp_resp uresp;
- int sqsize, rqsize;
+ unsigned int sqsize, rqsize;
  struct c4iw_ucontext *ucontext;
  int ret;
  struct c4iw_mm_entry *mm1, *mm2, *mm3, *mm4, *mm5 = NULL;
-- 
1.7.1